feat: add statusline component provider

neo451 opened the pull request:
If you use lualine: 
```lua
  {
    "nvim-lualine/lualine.nvim",
    opts = {
      sections = {
        lualine_x = {
          "g:obsidian",
        },
      },
    },
  },
```
the status is automatically updated through `vim.g.obsidian`

TODO: 
- [x] add in the readme/wiki.

sotte commented:
Can you post a before and after screenshot? @neo451

neo451 commented:
@sotte not much of a before and after, just emulates the obsidian app

![3814d017861999d874ec9f6b1ec29029](https://github.com/user-attachments/assets/0bb523e2-8966-4d05-8549-5d57c8286696)

![69e77b007d96f71d628b132c9c91dc15](https://github.com/user-attachments/assets/e21d6c2d-c5b5-47b1-8ee8-dcc2e03fbc3a)

sotte commented:
Nice.

A few thoughts:

- a screenshot of this should be added to the readme.
- nitpick: I would add another space between the sections to make it clear what the number belongs to what section.
- I think this should be optional / disabled by default. This would cause quite a few operations, right? An update every second, reading potentially 1000s of notes and every time you open a note. (I have a few thousands notes and would not use this.)

neo451 commented:
@sotte 
1. I will add the screenshot in the readme refactor, I plan to release it with `3.11.0` sometime in the next week.
2. I added the extra space like you mentioned
3. I added the extra checks before refresh, it now updates really lazily, only when you
1) open a note
2) note is different from the cached note, i.e. you jumped to new note or edited the frontmatter.

The other thing is that the thousands of notes are all read by rg, not neovim, and neovim just gets the response whenever rg finishes the work and updates. I think rg is powerful enough to read a thousand note, and normal laptops can spawn a few rgs without heating up, although I made the latter not likely to happen now as well.
